### DISCUSSION TITLE
Generator seeds for concatenating mediants

### DISCUSSION THREAD

**Discussion by id-entity** [geometor/relop#5] [General]
Some basic combinatorics done by hand, generator seeds so far up to 6 characters. Second number for how many mediant spaces in the seed. "C" before the number for closed mediant space in the center. 

Task: study and comparison and classification of how the seeds behave in relation to SB-interpretation. 

2a < > 			2b > <
< ```<>``` >			> ```><``` <
< ```<<>``` <> ```<>>``` >

4.1a ```<>``` ```<>```		   4.1b ```><``` ```><```
<> ```<><>``` <>		   >< ```><><``` ><
4.2a < ```<>``` > = 2a 	   4.2c < ```><``` >
			           < ```<><``` >< ```><>``` >
4.2b > ```<>``` <		    4.2d > >< < = 2b
> ```><>``` <> ```<><``` <
4.3a < < > >		   C4.3a < < > >
< << < <> > >> >	   < << < > >> >
4.3b < > < >		   C4.3b < > < >
< <> > >< < <> >	   < <> > < <> >
4.3c > < > <		   C4.3c > < > <
> >< < <> > >< <    > >< < > >< <
4.3d > > < <		   C4.3d > > < <
> >> > >< < << <    > >> > < << <

6.1a <<< >>>		6.2a < <<>> >		6.2i << <> >>
6.1b <<> <>>		6.2b < <><> >		6.2j << >< >>
6.1c <>> <<>		6.2c < >><< >		6.2k <> >< <>
6.1d <>< ><>		6.2d < ><>< >		6.2l <> <> <> = 4.1a
6.1e ><< >><		6.2e > <<>> <		6.2m >< <> ><
6.1f ><> <><		        6.2f > <><> <  	        6.2n >< >< >< = 4.1b
6.1g >>< ><<   	        6.2g > ><>< <		6.2o >> <> <<
6.1h >>> <<<		6.2h > >><< < 	        6.2p >> >< << 

6.3a < << >> >	6.3i << < > >>	C6.3a < << >> >	C6.3i << < > >>
6.3b < <> <> >	6.3j << > < >>	C6.3b < <> <> >	C6.3j << > < >>
6.3c < >> << >	6.3k <> > < <>	C6.3c < >> << >	C6.3k <> > < <>
6.3d < >< >< > 	6.3l <> < > <> 	C6.3d < >< >< > 	C6.3l <> < > <>
6.3e > << >> <	6.3m >< < > ><	C6.3e > << >> <	C6.3m >< < > ><
6.3f > <> <> <  	6.3n >< > < >< 	C6.3f > <> <> <  	C6.3n >< > < ><
6.3g > >< >< <	6.3o >> < > <<	C6.3g > >< >< <	C6.3o >> < > <<
6.3h > >> << < 	6.3p >> > < << 	C6.3h > >> << < 	C6.3p >> > < <<

6.4a < < <> > >	6.4b < < >< > >	6.4c < > >< < >	6.4d < > <> < > 	
6.4e > < <> > <	6.4f > < >< > <  	6.4g > > <> < <	6.4h > > >< < < 	

6.5a < < < > > >	C6.5a < < < > > >
6.5b < < > < > >	C6.5b < < > < > >
6.5c < > > < < >	C6.5c < > > < < >
6.5d < > < > < > 	C6.5d < > < > < >
6.5e > < < > > <	C6.5e > < < > > <
6.5f > < > < > <  	C6.5f > < > < > <
6.5g > > < > < <	C6.5g > > < > < <
6.5h > > > < < < 	C6.5h > > > < < <

**Comment by id-entity** [comment]
No possiblity to choose a better font to make concatenated <>  and >< more visible?

**Comment by phiarchitect** [comment]
I was going to ask if you could give me some examples - and here they are!

I suggest wrapping the blocks as code. In markdown, put three backticks "```" before and after the block and it will render as monospace - as such:

```
2a < > 2b > <
< <> > > >< <
< <<> <> <>> >
```
for inline code, use one backtick before and after - like so: `< <>< >< ><> >`

**Comment by phiarchitect** [comment]
I'd like to see a step by step narrative because I am not sure how the operations occur. Are you iterating over the set and performing transformations each time?

**Comment by id-entity** [comment]
Simplest mechanism in text editor
< > 
Copypaste '< > '
Insert copypaste after the mediant space and before right >
< < > >
Delete ' ' from the middle 
< ```<>``` >

Voila!

More generally, perhaps: start from a mediant space, read the words (as defined as characters between spaces) both left and right in memory. For a new row, copypaste the previous row and fill the mediants with new concatenations.

**Comment by phiarchitect** [comment]
I understand the operation for the second line below but not the third:
```
 2a  2b
 < > > <
 < <> > > >< <
 < <<> <> <>> >
```

**Comment by phiarchitect** [comment]
I am thinking about the logic of an interpreter reading these strings of symbols. 
what are the steps for a machine to run this?

what is starting to come to me is that the information and the instructions in this system are the same :)

**Comment by id-entity** [comment]
Bad formatting, sorry. 
2a
< >
< <> >
< <<> <> <>> >

This markup is gadawful...

**Comment by id-entity** [comment]
What are the steps for a machine that does what looks like copypasting for a writer? Is there contest of most efficient code of copypasting somewhere? 

Perhaps something like this: move cursor to beginning of a row. Read characters of two words into memory until second word separator. 
>Subroutine: Concatenate the words in memory into word by deleting their mediant space. 

<Move back to end of 1st word. Add space. Print word from memory. 

etc.

**Comment by id-entity** [comment]
That was an example of a crude linear and consecutive job. 

To do multilinear and parallel, posit multiple reading heads into each open mediant space of each linear row, bifurcate a reading head into two heads, which read simultaneously both left and ride and concatenate. Or something like that.

**Comment by id-entity** [comment]
You can also do first half a row and then the rest as the inverse that forms a palindrome. Many ways, thinking this as well we can sounds like a project of its own. What if we could use moebius tape? The possiblity of esolang hypercomputation angle messes my head... :D

**Comment by phiarchitect** [comment]
so let's create some pseudo code
first we have a string of symbols which we'll call `data`:
```js
var data = '< >'
```
next we have a function to perform the operation
```js
function operation(data) {
  // replace all '< >' with '< <> >'
  return  data.replace(/< >/g, '< <> >')
}
```
then call the function:
```js
data = operation(data)
```
if I run the function again, `data` will not change.

**Comment by id-entity** [comment]
That won't do in practice, global functions with preset data will create trouble. 

The basic algorithm in simple form needs to read toward its L(eft) and R(ight) into variables Ldata Rdata, and concatenate Ldata and Rdata into LdataRdata in the mediant cell " ".

If it helps, let's try think and write mediant cell as empty finite interval [ ]. 
<[ ]>
<[ ]<[ ]>[ ]>
Conc middle [ ] to []
<[ ]<[]>[ ]>
